src/pint/bayesian.py

```diff
@@ -0,0 +1,209 @@
+"""Bayesian interface providing the pulsar timing likelihood, prior and posterior functions."""
+
+from copy import deepcopy
+
+import numpy as np
+from scipy.stats import norm, uniform
+
+from pint.models.priors import Prior, UniformUnboundedRV
+from pint.residuals import Residuals
+
+
+class BayesianTiming:
+    """A wrapper around the PINT API that provides lnprior, prior_transform,
+    lnlikelihood, and lnposterior functions. This interface can be used to
+    draw posterior samples using the sampler of your choice.
+
+    Parameters
+    ----------
+    model : :class:`pint.models.timing_model.TimingModel`
+        Contains the input timing model. The best-fit values stored in this object
+        are not used.
+    toas : :class:`pint.toa.TOAs`
+        Contains the input toas.
+    use_pulse_numbers : bool, optional
+        How to handle phase wrapping. If True, will use the pulse numbers
+        from the toas object while creating :class:`pint.residuals.Residuals`
+        objects. Otherwise will use the nearest integer.
+    prior_info : dict, optional
+        A dict containing the prior information on free parameters. This parameter
+        supersedes any priors present in the model.
+
+    Notes
+    -----
+    1. The `prior` attribute of each free parameter in the `model` object should be set to
+       an instance of :class:`pint.models.priors.Prior`.
+
+    2. The parameters of BayesianTiming.model will change for every likelihood function call.
+       These parameters in general will not be the best-fit values. Hence, it is NOT a good
+       idea to save it as a par file.
+
+    3. Currently, only uniform and normal distributions are supported in prior_info. More
+       general priors should be set directly in the TimingModel object before creating the
+       BayesianTiming object. Here is an example prior_info object:
+       >>> prior_info = {

```

Comment:
This code block is still not formatting correctly for me at https://nanograv-pint--1407.org.readthedocs.build/en/1407/_autosummary/pint.bayesian.BayesianTiming.html.  Maybe add in a blank line and indent further?